### PR TITLE
91 enums from backend

### PR DESCRIPTION
**Pull request by potapchuk-anna**

### CODE REVIEW

File: backend/EasySpeak.Core/EasySpeak.Core.BLL/Interfaces/IConstantService.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+﻿using EasySpeak.Core.Common.DTO.Tag;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace EasySpeak.Core.BLL.Interfaces
+{
+    public interface IConstantService
```
Comment (Anatolii-Ivanov) [inline backend/EasySpeak.Core/EasySpeak.Core.BLL/Interfaces/IConstantService.cs:10]: IDataService or something like this

File: backend/EasySpeak.Core/EasySpeak.Core.BLL/Services/ConstantService.cs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+        public Task<List<TagWithImageDto>> GetTagsAsync()
+        {
+            return context.Tags.ProjectTo<TagWithImageDto>(mapper.ConfigurationProvider).ToListAsync();
+        }
+
+        public List<string> GetTimezones()
+        {
+            return GetEnumStringValues<Timezone>();
+        }
```
Comment (Anatolii-Ivanov) [inline backend/EasySpeak.Core/EasySpeak.Core.BLL/Services/ConstantService.cs:39]: I guess it's not needed

File: backend/EasySpeak.Core/EasySpeak.Core.DAL/Context/Seeder.cs [hunk 3]
```diff
@@ -168,15 +168,123 @@ private static IList<Lesson> GenerateLessons(int count = 10)
...
+                Id = 1,
+                Name = "Architecture",
+                ImageUrl = "ClassicalBuilding.svg",
+                CreatedAt = DateTime.Now,
+            },
+            new Tag
+            {
+                Id = 2,
+                Name = "Arts",
+                ImageUrl = "ArtistPalette.svg",
```
Comment (Anatolii-Ivanov) [inline backend/EasySpeak.Core/EasySpeak.Core.DAL/Context/Seeder.cs:186]: Can you create a list of tuples and then map it?

File: backend/EasySpeak.Core/EasySpeak.Core.DAL/Migrations/20230405093442_RemovedTagTable.cs [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
+            migrationBuilder.DropTable(
+                name: "LessonTag");
+
+            migrationBuilder.DropTable(
+                name: "TagUser");
+
+            migrationBuilder.DropTable(
+                name: "Tags");
```
Comment (Anatolii-Ivanov) [inline backend/EasySpeak.Core/EasySpeak.Core.DAL/Migrations/20230405093442_RemovedTagTable.cs:19]: Really... It's not the best idea to update data

File: frontend/src/app/core/services/constants.service.ts [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+interface Tag {
+    name: string;
+    imageUrl: string;
+}
+
+@Injectable({
+    providedIn: 'root',
+})
+export class ConstantsService {
```
Comment (Anatolii-Ivanov) [inline frontend/src/app/core/services/constants.service.ts:19]: DataService or something similar

File: frontend/src/app/modules/auth/sign-up/sign-up.component.ts [hunk 6]
```diff
@@ -89,7 +91,9 @@ export class SignUpComponent extends BaseComponent implements OnInit {
 
     private setUpData() {
         this.countries = this.countriesTzLangProvider.getCountriesList().map((x) => x.name);
-        this.languages = this.countriesTzLangProvider.getLanguagesList();
+        this.constantsService.getAllLanguages().subscribe((languages) => {
```
Comment (Anatolii-Ivanov) [inline frontend/src/app/modules/auth/sign-up/sign-up.component.ts:94]: Can we save it to localStorage and then check if there are languages there, use them, if not - load

**Review by Anatolii-Ivanov — CHANGES_REQUESTED** [review]
Have you updated tags too?

File: frontend/src/app/modules/auth/sign-up/sign-up.component.ts (hunk 6 above)
Comment (potapchuk-anna) [reply frontend/src/app/modules/auth/sign-up/sign-up.component.ts:94]: this logic is implemented here
frontend/src/app/core/services/constants.service.ts

File: backend/EasySpeak.Core/EasySpeak.Core.DAL/Migrations/20230405093442_RemovedTagTable.cs [hunk 7]
```diff
@@ -0,0 +1,178 @@
+﻿using System;
+using Microsoft.EntityFrameworkCore.Migrations;
+
+#nullable disable
+
+namespace EasySpeak.Core.DAL.Migrations
+{
+    public partial class RemovedTagTable : Migration
```
Comment (tatianahutii) [inline backend/EasySpeak.Core/EasySpeak.Core.DAL/Migrations/20230405093442_RemovedTagTable.cs:8]: This is not good practice to drop and recreate tables for data updating
You need to use SQL and write script for data cleaning

**Review by tatianahutii — DISMISSED** [review]

File: backend/EasySpeak.Core/EasySpeak.Core.DAL/Migrations/20230410003900_DuplicateTagsDeleted.cs [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+                    )
+                );
+
+                DELETE FROM Tags
+                WHERE Id NOT IN (
+                    SELECT MIN(Id)
+                    FROM Tags
+                    GROUP BY Name
+                );");
+        }
```
Comment (Anatolii-Ivanov) [inline backend/EasySpeak.Core/EasySpeak.Core.DAL/Migrations/20230410003900_DuplicateTagsDeleted.cs:48]: Won't it delete all tags except first one?

File: backend/EasySpeak.Core/EasySpeak.Core.DAL/Context/Seeder.cs [hunk 9]
```diff
@@ -138,11 +135,35 @@ private static IList<Lesson> GenerateLessons(int count = 10)
 
     private static IList<Tag> GenerateTags()
     {
-        return AllTags.Select((t, i) => new Tag
+        var tags = new[]
         {
-            Id = ++i,
-            Name = t,
-            CreatedAt = DefaultDate
+            ("Arts", "ArtistPalette.svg", new DateTime(2023, 04, 05)),
```
Comment (Anatolii-Ivanov) [inline backend/EasySpeak.Core/EasySpeak.Core.DAL/Context/Seeder.cs:140]: You can use DefaultDate, but ok

**Review by Anatolii-Ivanov — APPROVED** [review]
If you checked that your script is working, feel free to merge it

**Review by bochka123 — APPROVED** [review]

**Pull request merged**